### PR TITLE
config dbt docs generate GitHub Action

### PR DESCRIPTION
**Pull request by forgxyz**
# Description

_Please include a summary of changes and related issue (if any)._
Adds a GitHub Action script to generate docs and serve to a branch titled `docs`, from which we will host the docs on Netlify.
This means the end-analyst does not need to deal with generating docs on PR, they will be created and served with each merge thus always staying up to date.
  
I have added the credentials as repository secrets and tested implementation on a fork.

### CODE REVIEW

**Comment by forgxyz** [comment]
^ deploy checks will fail because I updated the config on Netlify to check the docs branch, which won't exist until this merges

File: .dbt/profiles.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      role: "{{ env_var('ROLE') }}"
+      user: "{{ env_var('USER') }}"
+      password: "{{ env_var('PASSWORD') }}"
+      region: "{{ env_var('REGION') }}"
+      database: "{{ env_var('DATABASE_PROD') }}"
+      warehouse: "{{ env_var('WAREHOUSE_PROD') }}"
+      schema: PROD
+      threads: 4
+      client_session_keep_alive: False
+      query_tag: gh_actions
```
Comment (antonyip) [inline .dbt/profiles.yml:29]: newline

File: .github/workflows/docs_update.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      - name: commit changed files
+        run: |
+          git config user.email "abc@xyz"
+          git config user.name "github-actions"
+          git commit -am "Auto-update docs"
+          
+      - name: push changes to docs
+        run: |
+          git push
```
Comment (antonyip) [inline .github/workflows/docs_update.yml:67]: newline

**Review by antonyip — APPROVED** [review]
nits and lets go

**Pull request merged**